### DISCUSSION TITLE
Create cuboid boundaries in a point cloud

### DISCUSSION THREAD

**Discussion by jkissi** [marcomusy/vedo#486] [Q&A] [answered]
Hi! 
I've recently discovered this library and am trying to get to grips with it...it seems like its got lots of great features.

I have a point cloud and I am trying to place bounding boxes around certain areas of the cloud, something like this:

![pyramids_edit](https://user-images.githubusercontent.com/8246894/137851152-89e7aa09-5b02-4dcd-97e9-50e5991e23dc.png)

Fairly simple I think but I'm not sure whats best to use. I think what I want is Unstructured Grid? Maybe something like the following:

```
    points = np.array([[1, 0, 0],
                          [0, 0, 0],
                          [0, 1, 0],
                          [1, 1, 0],
                          [0, 0, 1],
                          [1, 0, 1],
                          [0, 1, 1],
                          [1, 1, 1]])
    faces = np.array([1, 4, 6, 2, 0, 5, 7, 3])

    ug = vedo.UGrid([points, faces]).c('blue').alpha(0.5)
    vpts = vedo.Points(ug.points(), r=10).c('green')

    vedo.show(ug,
              vpts,
              vpts.labels('id'),
              axes=1)
```

I'm not sure how to call UGrid though. Can I use numpy arrays or dataframes to do it? And what are the arguments that it needs? I had a loo0k at the source but I wasn't quite clear what it was asking for. I like UGrid because I want transparency but to control the line thickness and colour. Any help would be very much appreciated!! 

Thanks in advance 

J

**Comment by marcomusy** [comment, marked as answer]
Hi, the UGrid is not what you need here.. you may play around with this:
```python
import vedo
import numpy as np

vedo.settings.useDepthPeeling = True # may improve rendering

points= np.array([[1, 0, 0],
                  [0, 0, 0],
                  [0, 1, 0],
                  [1, 1, 0],
])

line = vedo.Line(points, closed=True).lineWidth(5).c('red4')

wall = line.extrude(0.2)
#wall = line.sweep([0,0,0.2]) # also possbile
wall.c('blue4').alpha(0.5).lineWidth(0) # control apperance

labs = vedo.Points(points).labels('id')
#tube = vedo.Tube(points, r=.1)

vedo.show(line, wall, labs, axes=1)
```
![Screenshot from 2021-10-19 12-19-46](https://user-images.githubusercontent.com/32848391/137891231-885a3bf8-e5d2-43a5-a0b1-684d100496fa.png)

**Reply by jkissi** [reply]
Hi Marco!! 
First off, let me thank you for responding so quickly to me. I am currently developing a big algorithm for my research, so I'm very appreciative of the fast response you've given here. Thanks. 

The solution you offered was interesting but didn't fit the best with the way my algorithm works: it provides me with spatial coordinates for the boundaries of the boxes, so drawing a single line and extruding, doesn't currently work for my purpose. But, you inspired me to look a little deeper and I came up with something based on what you replied to me: 

```
    lines = [[0, 1], [1, 2], [2, 3], [3, 0], [4, 5], [5, 6],
             [6, 7], [7, 4], [0, 4], [1, 5], [2, 6], [3, 7]]

    points = np.array([[1, 0, 0],
                       [0, 0, 0],
                       [0, 1, 0],
                       [1, 1, 0],
                       [1, 0, 1],
                       [0, 0, 1],
                       [0, 1, 1],
                       [1, 1, 1]])

    line_array = []
    for i in np.arange(len(lines)):
        line = vedo.Line(points[lines[i]], closed=True).lineWidth(5).c('green')
        line_array.append(line)

    labs = vedo.Points(points).labels('id')
    vedo.show(line_array, labs, axes=1)
```
That results in this: 

![vedo_cube_edit](https://user-images.githubusercontent.com/8246894/138014046-6204e00d-bcfe-47b9-aa21-e1d5b0a9c325.png)


This is basically what I'm after. Thanks for your help! I'll probably be back soon with more questions, so looking forward to further communications!! 

Cheers
J

**Reply by marcomusy** [reply]
hi, consider you also have this functionality:
```python
from vedo import *

s = Points(Sphere())

print(s.bounds())

box = s.box().lw(4).c('red3')

show(s, box, axes=1)
```
<img width="309" alt="Screenshot 2021-10-20 at 10 31 31" src="https://user-images.githubusercontent.com/32848391/138057778-3f20374b-1198-40df-ac25-b2db905a8102.png">

**Reply by jkissi** [reply]
Very nice!! thanks for the heads up!
